Example use cases of programmatically managing access requests (#1165)
huggingface/hub-docs: docs/hub/models-gated.md

The base URL for the HTTP endpoints above is `https://huggingface.co`.

Those endpoints are not officially supported in `huggingface_hub` or `huggingface.js` yet but [this code snippet](https://github.com/huggingface/huggingface_hub/issues/1535#issuecomment-1614693412) (in Python) might help you getting started.

> NEW! There's an [open PR](https://github.com/huggingface/huggingface_hub/pull/1905) in `huggingface_hub` to add official support from our Python library.

Reviewer: This is merged now :)

Author: yep i know but @Wauplin is too fast 💨

Reviewer 2: I'm fine with opening a new PR once it's officially released so that we don't block this doc PR in the meantime (it's a matter of a few days). So fine to keep it like this for now :)

> NEW! There's an [open PR](https://github.com/huggingface/huggingface_hub/pull/1905) in `huggingface_hub` to add official support from our Python library.

Author: ```suggestion
NEW! There's an [merged PR](https://github.com/huggingface/huggingface_hub/pull/1905) in `huggingface_hub` to add official support from our Python library.
```

---
```

### Example use cases of programmatically managing access requests

Here are a few interesting use cases of programmatically managing access requests for gated repos we've seen organically emerge in the community.
As a reminder, the model repo needs to be set to manual approval, otherwise users get access to it automatically.

Possible use cases of programmatic management include:

> - If you have advanced user request screening requirements (for advanced compliance requirements, etc) or you wish to handle the user requests outside the Hub.

Reviewer 2: ```suggestion
- If you have advanced user request screening requirements (for compliance requirements, etc) or you wish to handle the user requests outside the Hub.
```

I find this sentence quite heavy to read (and a bit redundant)

    - An example for this was Meta's [Llama 2](https://huggingface.co/meta-llama/Llama-2-7b-chat-hf) initial release where users had to request access on a Meta website.
    - You can ask users for their HF username in your access flow, and then use a script to programmatically accept user requests on the Hub based on your set of conditions.
- If you want to condition access to a model based on completing a payment flow (note that the actual payment flow happens outside of the Hub).
    - Here's an [example repo](https://huggingface.co/Trelis/openchat_3.5-function-calling-v3) from TrelisResearch that uses this use case.

>    - [@RonanMcGovern](https://huggingface.co/RonanMcGovern) has posted a [video about the flow](https://twitter.com/TrelisResearch/status/1729227883296522338) and tips on how to implement it.

Author: ```suggestion
   - [@RonanMcGovern](https://huggingface.co/RonanMcGovern) has posted a [video about the flow](https://www.youtube.com/watch?v=2OT2SI5auQU) and tips on how to implement it.
```
